diff -ruN old-gmp-4.3.2/mpz/inp_str.c gmp-4.3.2/mpz/inp_str.c
--- old-gmp-4.3.2/mpz/inp_str.c	2010-01-07 17:09:03.000000000 -0300
+++ gmp-4.3.2/mpz/inp_str.c	2019-12-26 02:36:09.511624294 -0300
@@ -31,6 +31,9 @@
 extern const unsigned char __gmp_digit_value_tab[];
 #define digit_value_tab __gmp_digit_value_tab
 
+extern size_t
+mpz_inp_str_nowhite (mpz_ptr x, FILE *stream, int base, int c, size_t nread);
+
 size_t
 mpz_inp_str (mpz_ptr x, FILE *stream, int base)
 {
