diff -ruN old-gcc-6.4.0/gcc/system.h gcc-6.4.0/gcc/system.h
--- old-gcc-6.4.0/gcc/system.h	2019-12-19 23:11:27.466735175 -0300
+++ gcc-6.4.0/gcc/system.h	2020-01-08 22:14:16.317892612 -0300
@@ -484,7 +484,7 @@
 extern char *getwd (char *);
 #endif
 
-#if defined (HAVE_DECL_SBRK) && !HAVE_DECL_SBRK
+#if 0
 extern void *sbrk (int);
 #endif
 
@@ -537,7 +537,7 @@
 #if !defined (HAVE_STRSIGNAL) \
     || (defined (HAVE_DECL_STRSIGNAL) && !HAVE_DECL_STRSIGNAL)
 # ifndef strsignal
-extern const char *strsignal (int);
+extern char *strsignal (int);
 # endif
 #endif
 
diff -ruN old-gcc-6.4.0/gcc/timevar.c gcc-6.4.0/gcc/timevar.c
--- old-gcc-6.4.0/gcc/timevar.c	2019-12-19 23:11:26.986735199 -0300
+++ gcc-6.4.0/gcc/timevar.c	2019-12-27 22:57:23.715906619 -0300
@@ -24,6 +24,8 @@
 #include "timevar.h"
 #include "options.h"
 
+#define HAVE_STRUCT_TMS 1
+
 #ifndef HAVE_CLOCK_T
 typedef int clock_t;
 #endif
diff -ruN old-gcc-6.4.0/libiberty/unlink-if-ordinary.c gcc-6.4.0/libiberty/unlink-if-ordinary.c
--- old-gcc-6.4.0/libiberty/unlink-if-ordinary.c	2019-12-19 23:11:18.848735608 -0300
+++ gcc-6.4.0/libiberty/unlink-if-ordinary.c	2019-12-29 03:33:16.100444276 -0300
@@ -50,6 +50,14 @@
 
 #include "libiberty.h"
 
+/*
+ * Nanvix do not have lstat syscall, so lets undef
+ * the following and binutils will nicely use stat
+ * instead ;-).
+ */
+#undef S_ISLNK
+#undef S_IFLNK
+
 #ifndef S_ISLNK
 #ifdef S_IFLNK
 #define S_ISLNK(m) (((m) & S_IFMT) == S_IFLNK)
